File: examples/conditions-operator-all-values/conditions-operator-all-values.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Operator, Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.Dynamodb()
      .allow()
      .toGetItem()
      .onTable('Thread')
      .ifAttributes(
        ['ID', 'Message', 'Tags'],
        new Operator().stringEquals().forAllValues(),
      )
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);

File: examples/conditions-operator-all-values/conditions-operator-all-values.py
from iam_floyd import statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.Dynamodb() \
        .allow() \
        .to_get_item() \
        .on_table('Thread') \
        .if_attributes(['ID', 'Message', 'Tags'],
                       statement.Operator().string_equals().for_all_values())
    # doc-end
)

all = [s]
out(all)
deploy(all)
